crates/fluvio-hub-util/src/htclient.rs

```diff
@@ -1,13 +1,85 @@
+use std::fmt::Display;
+
 pub use http;
 pub use http::StatusCode;
 pub use http::{Request, Response};
 
 use anyhow::{anyhow, Result};
+use percent_encoding::{utf8_percent_encode, AsciiSet, CONTROLS};
 use serde::de::DeserializeOwned;
 
 #[cfg(not(target_arch = "wasm32"))]
 use ureq::OrAnyStatus;
 
+/// Special characters needing encoding are the characters below.
+/// Other characters don't need to be encoded, though they could.
+///
+/// https://developer.mozilla.org/en-US/docs/Glossary/Percent-encoding
+const QUERY_PERCENT_ENCODE_SET: &AsciiSet = &CONTROLS
+    .add(b':')
+    .add(b'/')
+    .add(b'?')
+    .add(b'#')
+    .add(b'[')
+    .add(b']')
+    .add(b'@')
+    .add(b'!')
+    .add(b'$')
+    .add(b'&')
+    .add(b'\'')
+    .add(b'(')
+    .add(b')')
+    .add(b'*')
+    .add(b'+')
+    .add(b',')
+    .add(b';')
+    .add(b'=')
+    .add(b'%')
+    .add(b' ')
+    .add(b'+');
+
+/// URL Query Parameters manager with support for encoding of special characters
+/// using UTF-8 Percent Encoding.
+#[derive(Debug)]
+pub struct UrlQueryParams(Vec<(String, String)>);
+
+impl Default for UrlQueryParams {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl UrlQueryParams {
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn insert<T: ToString>(&mut self, key: T, value: T) {
+        self.0.push((key.to_string(), value.to_string()));
+    }
+}
+
+impl Display for UrlQueryParams {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
we already have the url crate, could we use the url. form_urlencoded serializer/deserializer instead of reimplmenting this?  Though if we don't inject them to the artifact download, none of this may be needed?

Reply:
Of course! Will take a look on form_urlencoded!